Gems/ONNX/Code/Source/Clients/Model.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/TickBus.h>
+#include <AzCore/Debug/Timer.h>
+#include <AzFramework/StringFunc/StringFunc.h>
+#include <ONNX/ONNXBus.h>
+#include <onnxruntime_cxx_api.h>
+
+#include "upng/upng.h"
+
+#include <algorithm>
+#include <array>
+#include <filesystem>
+#include <string>
+
+namespace ONNX
+{
+    //! Generic ONNX model which can be used to create an inference session and run inferences.
+    class Model
+    {
+    public:
+        Model()
+        {
+        }
+        //! Required params to create session and run inference, passed to Load().
+        struct InitSettings
+        {
+            //! Source of onnx model file.
+            std::wstring m_modelFile = std::wstring{ W_GEM_ASSETS_PATH } + std::wstring{ L"/model.onnx" };
+            std::string m_modelName = ""; //!< Used to create groupings for ImGui dashboard graphs in editor, idea is that the inference runtimes from the same model instance get displayed on the same graph.
+            std::vector<int64_t> m_inputShape; //!< Specifies dimensions of input, eg a vector specifying dimension and magnitude of dimension such as { 1, 1, 28, 28 }.
+            std::vector<int64_t> m_outputShape; //!< Specifies dimensions of output, eg a vector specifying dimension and magnitude of dimension such as { 1, 10 }.
+            bool m_cudaEnable = false; //!< Toggle to create a CUDA session on gpu, if disabled normal cpu session created.
+        };
+        //! Initialises necessary params in order to run inference.
+        //! Must be executed before Run().
+        //! Creates the session, memory info, and extracts input and output names and count from onnx model file.
+        //! Only needs to be executed once, inferences using the same onnx model file can be run by providing different input/output params to
+        //! Run().
+        void Load(InitSettings& m_init_settings);
```

Comment:
Nit: const InitSettings&

Reply:
Also per the coding standard, the name of local variables should not start with `m_` and doesn't should be lowerCamelCase.

So the parameter variable should be named something such as "initSettings"

https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#variable-names